desktop/AutoStartManager.ts

```diff
@@ -62,10 +67,19 @@ class AutoStartManager {
     );
   }
 
-  static isEnabled = () => AutoStartManager.service.isEnabled();
+  static restoreNativeAutoLaunchStatus = async () => {
+    try {
+      const status = await AutoStartManager.service.isEnabled();
+      StoreService.set(IS_AUTO_START_ENABLED, status);
```

Comment:
**Question:**

I have a question here:

Now the release version works on `isEnabled` function, and only this function determines whether applications will be launched on the system start or not. If I make a reversal of this change, we will only rely on the previous config. This config will not provide any information about the script installed in the system that runs the application.

Based on this, we can only not ask the system at startup and not show the system window if it does not require it. But we cannot rely only on a constant.

As it was [before](https://github.com/spacemeshos/smapp/commit/763c6521ddf602528bcdab4d5414bbb418a3d432#diff-9220b20ab5609061877b08603e7d7ea5de9b136ec915953dbc5f7773600a2a77R26):

```javascript
if (StoreService.get(IS_AUTO_START_ENABLED)) {
    await this.manager.enable();
}
```

In this previous code snippet, it may seem like we're forcing the system to enable auto-start, but we're not effectively managing this process. The isEnabled function precisely checks both the script's presence and the necessary permissions for running the app during system startup.

So, even though we're enabling the system, it doesn't ensure proper management of this process. We need to consider a more balanced approach to maintain functionality while controlling system startup behavior.

Reply:
There was a bug. Should be:
```
if (StoreService.get(IS_AUTO_START_ENABLED)) {
  await this.manager.enable();
} else {
  await this.manager.disalbe();
}
```

And, ideally, we need to handle permission issues by showing an error popup with a proper message.

Reply:
I'm not entirely clear on why we need this code in the initialization process. We can rely on the constant only at the beginning of the system, based on the saved value. The script for the auto-launch is reused from the previous version of the app. Why do we need to duplicate the logic, add another script, and also depend on the constant value? Are our users using a config.json file and configuring it manually? Maybe I'm missing something here.

I clearly see why we need the isEnabled -> store the same value in the config.json. But really on the config.json to add or remove the auto-launch script, I don't understand.

Reply:
Answer these questions:
1. Why do we need this value to be stored in config.json?
2. Since we need it in config.json — User may want to change something here (even manually), what then?
3. Also User may want to copy-paste contents of AppData directory from one PC to another to have absolutely the same everything in Smapp, what then?